test/utils/wabt_engine.cpp

```diff
@@ -37,123 +49,126 @@ std::unique_ptr<WasmEngine> create_wabt_engine()
     return (result == wabt::Result::Ok);
 }
 
 bool WabtEngine::instantiate(bytes_view wasm_binary)
 {
-    wabt::interp::HostModule* hostModule = m_env.AppendHostModule("env");
-    assert(hostModule != nullptr);
-
-    hostModule->AppendFuncExport("adler32", {{wabt::Type::I32, wabt::Type::I32}, {wabt::Type::I32}},
-        [this](const wabt::interp::HostFunc*, const wabt::interp::FuncSignature*,
-            const wabt::interp::TypedValues& args, wabt::interp::TypedValues& results) {
-            const auto offset = args[0].value.i32;
-            const auto length = args[1].value.i32;
+    wabt::Errors errors;
+    wabt::interp::ModuleDesc module_desc;
+    const wabt::Result result = wabt::interp::ReadBinaryInterp(
+        wasm_binary.data(), wasm_binary.size(), wabt::ReadBinaryOptions{}, &errors, &module_desc);
+    if (result != wabt::Result::Ok)
+        return false;
+
+    m_module = wabt::interp::Module::New(m_store, module_desc);
+
+    auto host_func = wabt::interp::HostFunc::New(m_store,
+        wabt::interp::FuncType{{wabt::Type::I32, wabt::Type::I32}, {wabt::Type::I32}},
+        [](wabt::interp::Thread& thread, const wabt::interp::Values& args,
+            wabt::interp::Values& results, wabt::interp::Trap::Ptr*) -> wabt::Result {
+            const auto offset = args[0].i32_;
+            const auto length = args[1].i32_;
             // TODO: move the memory lookup outside in order to reduce overhead
-            assert(m_env.GetMemoryCount() > 0);
-            auto memory = m_env.GetMemory(0);
-            assert(memory != nullptr);
-            auto memory_data = memory->data;
-            assert(memory_data.size() > (offset + length));
+            assert(!thread.GetCallerInstance()->memories().empty());
+            wabt::interp::Memory::Ptr memory(
+                thread.store(), thread.GetCallerInstance()->memories()[0]);
+            assert(memory);
+            auto memory_data = memory->UnsafeData();
+            assert(memory->ByteSize() > (offset + length));
             const auto ret =
                 fizzy::test::adler32({reinterpret_cast<uint8_t*>(&memory_data[offset]), length});
-            results[0].set_i32(ret);
-            return wabt::interp::Result::Ok;
+            results[0].Set(ret);
+            return wabt::Result::Ok;
         });
 
-    wabt::Errors errors;
-    const wabt::Result result = wabt::ReadBinaryInterp(&m_env, wasm_binary.data(),
-        wasm_binary.size(), wabt::ReadBinaryOptions{}, &errors, &m_module);
-    if (result != wabt::Result::Ok)
-        return false;
-    assert(m_module != nullptr);
+    wabt::interp::Trap::Ptr trap;
+    m_instance =
+        wabt::interp::Instance::Instantiate(m_store, m_module.ref(), {host_func->self()}, &trap);
 
-    // Run start function (this will return ok if no start function is present)
-    const wabt::interp::ExecResult r = m_executor.RunStartFunction(m_module);
```

Comment:
Is this now executed properly by instantiate?

Reply:
Yes, looks like so, our unit test for this passes.